extensions/big-data-cluster/src/bigDataCluster/dialog/mountHdfsDialog.ts

```diff
@@ -0,0 +1,401 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as vscode from 'vscode';
+import * as azdata from 'azdata';
+import * as nls from 'vscode-nls';
+import { ClusterController, ControllerError, MountInfo, MountState } from '../controller/clusterControllerApi';
+import { AuthType } from '../constants';
+
+const localize = nls.loadMessageBundle();
+
+const basicAuthDisplay = localize('basicAuthName', "Basic");
+const integratedAuthDisplay = localize('integratedAuthName', "Windows Authentication");
+
+function getAuthCategory(name: AuthType): azdata.CategoryValue {
+	if (name === 'basic') {
+		return { name: name, displayName: basicAuthDisplay };
+	}
+	return { name: name, displayName: integratedAuthDisplay };
+}
+
+/**
+ * Converts a comma-delimited set of key value pair credentials to a JSON object.
+ * This code is taken from the azdata implementation written in Python
+ */
+function convertCredsToJson(creds: string): { credentials: {} } {
+	if (!creds) {
+		return undefined;
+	}
+	// TBD: can we use something other than negative lookup?
+	let credObj = { 'credentials': {} };
+	let pairs = creds.split(',');
+	// handle escaped commas in a browser-agnostic way by pushing to the next
+	let validPairs: string[] = [];
+	for (let i = 0; i < pairs.length; i++) {
+		if (i < (pairs.length - 1) && pairs[i].match(/(?!\\).*\\$/)) {
+			// Since this was meant to be an escaped comma, restore it
+			pairs[i + 1] = `${pairs[i]},${pairs[i + 1]}`;
+		} else {
+			validPairs.push(pairs[i]);
+		}
+	}
+
+	validPairs.forEach(pair => {
+		const formattingErr = localize('mount.err.formatting', "Bad formatting of credentials at {0}", pair);
+		try {
+			// # remove escaped characters for ,
+			pair = pair.replace('\\,', ',').trim();
+			let firstEquals = pair.indexOf('=');
+			if (firstEquals <= 0 || firstEquals >= pair.length) {
+				throw new Error(formattingErr);
+			}
+			let key = pair.substring(0, firstEquals);
+			let value = pair.substring(firstEquals + 1);
+			credObj.credentials[key] = value;
+		} catch (err) {
```

Comment:
the actual err object is not being used, by design?

Reply:
Correct. Again, taken from Python but the concept is any error indicates it isn't in key-value-pair format